pkg/operation/botanist/controlplane/kube_apiserver_service.go

```diff
@@ -125,16 +149,13 @@ func (d *kubeAPIService) Wait(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
 	defer cancel()
 
-	svcNS := d.service.Namespace
-	svcName := d.service.Name
-
-	if d.values.EnableSNI {
-		svcNS = d.sniService.Namespace
-		svcName = d.sniService.Name
-	}
-
 	return d.waiter.Until(ctx, 5*time.Second, func(ctx context.Context) (done bool, err error) {
-		loadBalancerIngress, err := kutil.GetLoadBalancerIngress(ctx, d.client, svcNS, svcName)
+		loadBalancerIngress, err := kutil.GetLoadBalancerIngress(
```

Comment:
Can you add a comment that it is either the Shoot API server service or the istio ingress gateway service - based on the SNI phase (enabling, disabling, ...) ? 
For me that is not obvious right away due to the changes you made.